pkgs/tools/misc/fzf/default.nix

```diff
@@ -1,12 +1,27 @@
-{ lib
+{ stdenv
+, lib
+, pkgs
 , buildGoModule
 , fetchFromGitHub
 , writeText
 , runtimeShell
 , installShellFiles
 , ncurses
 , perl
+, glibcLocales
 }:
+
+let
+  # on Linux, wrap perl in the bash completion scripts with the glibc locales,
+  # so that using the shell completion (ctrl+r, etc) doesn't result in ugly
+  # warnings on non-nixos machines
+  ourPerl = if stdenv.isDarwin then perl else (
+    pkgs.writers.writeBashBin "perl" ''
```

Comment:
Why is `pkgs` used here and on the following line?

Reply:
Also why use `writeBashBin` (which only seems be used in a couple of places in the repo) + `runtimeShell` instead of `writeShellScript`?